lib/drivers/medtronic600/NGPHistoryParser.js

```diff
@@ -1566,29 +1562,34 @@ class NGPHistoryParser {
     if (lastTempBasalEvent &&
       lastTempBasalEvent.eventType === NGPHistoryEvent.EVENT_TYPE.TEMP_BASAL_PROGRAMMED) {
       const suppressedBasal = this.findSuppressedBasal(lastTempBasalEvent);
-      const basal = this.cfg.builder.makeTempBasal()
-        .with_rate(lastTempBasalEvent.rate)
-        .with_expectedDuration(lastTempBasalEvent.expectedDuration)
-        .with_duration(lastTempBasalEvent.expectedDuration)
-        .set('suppressed', {
-          type: 'basal',
-          deliveryType: 'scheduled',
-          rate: suppressedBasal.rate,
-          scheduleName: suppressedBasal.patternName,
-        });
+      // Don't create the temp basal if we don't find the underlying suppressed info
+      // This could happen if the ONLY basal event we retrieve from the pump history
+      // is a TEMP_BASAL_PROGRAMMED.
+      if (!_.isUndefined(suppressedBasal)) {
```

Comment:
How about still displaying the temp basal, but not adding the suppressed info if not available? Suppressed info is an optional field.

Reply:
I have confirmed that uploading a temp basal without a suppressed basal after uploading the same temp basal that includes the suppressed information will mark the basal with the suppressed info as inactive. This is not what we want. So either we need to modify the deduplicator to prefer basals with suppressed info, or keep this uploader code as-is.

Reply:
OK, let's keep it as-is for the moment.